02 - JS and CSS Clock/index.html

```diff
@@ -62,13 +62,34 @@
       background: black;
       position: absolute;
       top: 50%;
+      transform-origin: 100%;
+      transform: rotate(90deg);
+      transition: all .5s ease-in-out;
     }
 
   </style>
 
   <script>
+    const initialRotate = 90;
+    setInterval( timeTick, 1000 );
+    timeTick();
 
+    function timeTick() {
+      const time = new Date();
+      const seconds = time.getSeconds();
+      const minutes = time.getMinutes();
+      const hours = time.getHours();
 
+      setHandRotation( "second-hand", seconds, 60 );
+      setHandRotation( "min-hand", minutes, 60 );
+      setHandRotation( "hour-hand", hours, 12 );
+    }
+
+    function setHandRotation( handClass, currentTimeValue, maxTimeValue ) {
+      const hand = document.querySelector(`.hand.${handClass}`);
+      const handRotation = ( currentTimeValue / maxTimeValue ) * 360;
+      hand.style.transform = `rotate(${handRotation + initialRotate}deg)`;
+    }
```

Comment:
The logic checks out, although this makes it so that three new query selector happens for every tick, which is an expensive operation.

The number of hands on the clock is fixed, and there's no need to select them dynamically every time. An approach to consider here would be to cache the elements for the three hands to three variables beforehand, and just use those variables for every tick.

Reply:
I did this to lessen the lines of code but didn't expect that this is more expensive in the long run. Got this!
Pushed the fix for both this issue and the merge conflict.